### PR TITLE
chore: remove hashing function

### PR DESCRIPTION
**Pull request by velkoskis**
## What does this implement/fix? Explain your changes.
This removes the existing hash function from the auth logic . It serves the purpose of simplifying and using an .env variable as the auth token for the proposer routes.
## Does this close any currently open issues?
no
## What commands can I run to test the change? 
npm run test-e2e-protocol            
## Any other comments?
It is suggested for the deployer to use UUID(https://www.uuidgenerator.net/) as the authentication token. However, the code doesn't stop the deployer to use whatever string they want.

Updated administrator and circuits tests.

### CODE REVIEW

File: cli/lib/nf3.mjs [hunk 1]
```diff
@@ -107,10 +107,7 @@ class Nf3 {
     this.ethereumSigningKey = ethereumSigningKey;
     this.zkpKeys = zkpKeys;
     this.currentEnvironment = environment;
-    axios.defaults.headers.common['X-APP-TOKEN'] = crypto
-      .createHash('sha256')
-      .update(environment.PROPOSER_KEY)
-      .digest('hex');
+    axios.defaults.headers.common['X-APP-TOKEN'] = environment.PROPOSER_KEY;
```
[the remark this answers is not in the document]
Comment (velkoskis) [reply cli/lib/nf3.mjs:110]: Got it! I am going to update this. I misunderstood.

[the remark this answers is not in the document]
Comment (velkoskis) [reply cli/lib/nf3.mjs]: @druiz0992 Do we have a preference if this random content should be hashed in any way or it is okay to be just a string that the user configures (preferably uuid)?

[the remark this answers is not in the document]
Comment (druiz0992) [reply cli/lib/nf3.mjs]: String is ok

File: config/default.js [hunk 2]
```diff
@@ -187,6 +187,7 @@ module.exports = {
         process.env.CHALLENGER_KEY ||
         process.env.BOOT_CHALLENGER_KEY ||
         '0xd42905d0582c476c4b74757be6576ec323d715a0c7dcff231b6348b7ab0190eb',
```
Comment (druiz0992) [inline config/default.js:189]: Why is it defined as part of the environment? If you maintain it there, can you also add an entry in aws?

File: nightfall-optimist/src/utils/auth.mjs [hunk 3]
```diff
@@ -1,11 +1,5 @@
...
-import crypto from 'crypto';
-
-const environment = config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
-
 export default function auth(req, res, next) {
-  const token = req.headers['x-app-token'];
-  if (token === crypto.createHash('sha256').update(environment.PROPOSER_KEY).digest('hex'))
-    return next();
+  const token = req.headers['X-APP-TOKEN'];
+  if (token === process.env.AUTH_TOKEN) return next();
```
Comment (druiz0992) [inline nightfall-optimist/src/utils/auth.mjs:3]: Should we take this from config?

File: config/default.js (hunk 2 above)
Comment (velkoskis) [reply config/default.js:189]: Yes, let's add it to aws and polygonEdge!

File: nightfall-optimist/src/utils/auth.mjs (hunk 3 above)
Comment (velkoskis) [reply nightfall-optimist/src/utils/auth.mjs:3]: no, if we maintain it in the environments, we need to call it from there.

**Review by daveroga — APPROVED** [review]

**Review by imagobea — APPROVED** [review]
Thank you for taking care of this :raised_hands:

**Pull request merged**